### PR TITLE
feature: [vmagent] Add service discovery support for Vultr

### PR DESCRIPTION
**Pull request by jiekun**
### Describe Your Changes
related issue: https://github.com/VictoriaMetrics/VictoriaMetrics/issues/6041

#### Added
- Added service discovery support for Vultr.

#### Docs
- `CHANGELOG.md`, `sd_configs.md`, `vmagent.md` are updated.

#### Note
- Useful links: 
    - Vultr API: https://www.vultr.com/api/#tag/instances/operation/list-instances
    - Vultr client SDK: https://github.com/vultr/govultr
    - Prometheus SD: https://github.com/prometheus/prometheus/tree/main/discovery/vultr

---
### Checklist

The following checks are mandatory:

- [X] I have read the [Contributing Guidelines](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/CONTRIBUTING.md)
- [x] All commits are signed and include `Signed-off-by` line. Use `git commit -s` to include `Signed-off-by` your commits. See this [doc](https://git-scm.com/book/en/v2/Git-Tools-Signing-Your-Work) about how to sign your commits.
- [x] Tests are passing locally. Use `make test` to run all tests locally.
- [x] Linting is passing locally. Use `make check-all` to run all linters locally.

Further checks are optional for External Contributions:

- [X] Include a link to the GitHub issue in the commit message, if issue exists.
- [x] Mention the change in the [Changelog](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/docs/CHANGELOG.md). Explain what has changed and why. If there is a related issue or documentation change - link them as well.

  Tips for writing a good changelog message::

    * Write a human-readable changelog message that describes the problem and solution.
    * Include a link to the issue or pull request in your changelog message.
    * Use specific language identifying the fix, such as an error message, metric name, or flag name.
    * Provide a link to the relevant documentation for any new features you add or modify.

- [ ] After your pull request is merged, please add a message to the issue with instructions for how to test the fix or try the feature you added. Here is an [example](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4048#issuecomment-1546453726)
- [x] Do not close the original issue before the change is released. Please note, in some cases Github can automatically close the issue once PR is merged. Re-open the issue in such case.
- [x] If the change somehow affects public interfaces (a new flag was added or updated, or some behavior has changed) - add the corresponding change to documentation.

### CODE REVIEW

**Comment by jiekun** [comment]
### Note for reviewers
- Just encountered some network issue so I closed the previous PR and re-opened a new one. They are identical.
- In prometheus which use Vultr client SDK, ipv6 related fields are defined as follow:

```go
	V6Network        string `json:"v6_network"`
	V6MainIP         string `json:"v6_main_ip"`
	V6NetworkSize    int    `json:"v6_network_size"`
```
 
- However, in Vultr API doc (see the link above), ipv6 networks is an array with struct:

```go
        ...
        V6Networks       []V6Networks `json:"v6_networks"`
        ...

type V6Networks struct {
	Network     string `json:"network"`
	MainIP      string `json:"main_ip"`
	NetworkSize string `json:"network_size"`
}
```

In this PR the ipv6 fields are defined as Prometheus does, which is the same as real-world response from Vultr API (I tested it with a Vultr account / VPS).

**Comment by jiekun** [comment]
@f41gh7 I just rebased my branch, but the coverage for `codecov/project` is still below the threshold. May I seek your suggestion on how to fix it? The coverage of the code changed in this PR is already high, tho. 

And I think the PR is ready for review :)

**Comment by hagen1778** [comment]
@f41gh7 @AndrewChubatiuk would appreciate review from your side.

File: lib/promscrape/discovery/vultr/instance.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	GatewayV4        string `json:"gateway_v4"`
+	V6Network        string `json:"v6_network"`
+	V6MainIP         string `json:"v6_main_ip"`
+	V6NetworkSize    int    `json:"v6_network_size"`
+	Hostname         string `json:"hostname"`
+	Label            string `json:"label"`
+	// Deprecated: Tag should no longer be used. Instead, use Tags.
+	Tag             string   `json:"tag"`
+	InternalIP      string   `json:"internal_ip"`
+	KVM             string   `json:"kvm"`
```
Comment (AndrewChubatiuk) [inline lib/promscrape/discovery/vultr/instance.go:43]: i think it makes sense to comment out unused fields

**Comment by AndrewChubatiuk** [comment]
LGTM

File: lib/promscrape/discovery/vultr/instance.go (hunk 1 above)
Comment (jiekun) [reply lib/promscrape/discovery/vultr/instance.go]: Good catch! Thank you for your review. The unused fields have been commented out.

**Review by f41gh7 — APPROVED** [review]
LGTM

**Comment by jiekun** [comment]
Thank you for the review. I could fix those conflicts when this PR is ready to be merged into upstream.

**Comment by f41gh7** [comment]
@jiekun  could you please squash and sign all commits?

**Comment by jiekun** [comment]
> @jiekun could you please squash and sign all commits?

Done. The linter failed due to the latest release of Go (1.22.3), which contains a security fix.

**Comment by f41gh7** [comment]
Thanks for contribution!

**Comment by valyala** [comment]
@jiekun , @f41gh7 , please see the follow-up commit 35b3b95cbc623a58642f50151bf533d2fd4f7bca

**Pull request merged**